[front] - chore(AttachmentPicker): use DropdownMenuFilters

JulesBelveze opened the pull request:
## Description

This PR upgrades `@dust-tt/sparkle` to `0.4.24` and refactors the `InputBarAttachmentsPicker` to use the new `DropdownMenuFilters` component instead of custom filter buttons. The new component provides a more consistent and maintainable approach to displaying filter chips for data sources and tools in the attachments picker.

## Risk
Low - UI refactoring using a new Sparkle component, with library upgrade from 0.4.23 to 0.4.24.

## Tests
Manual testing of the attachment picker filtering UI.

## Deploy Plan
* Publish Sparkle 0.4.24
* Deploy front and extension

JulesBelveze committed:
[extension] - fix: update @dust-tt/sparkle to version 0.4.24

 - Ensure package-lock.json and package.json reflect the updated version number
 - Resolve potential security vulnerabilities or bugs in the previous version

JulesBelveze committed:
[front/components/assistant/conversation/input_bar] - refactor: replace manual source filter buttons with DropdownMenuFilters component

- Implemented DropdownMenuFilters to manage source selection in the attachments picker UI
- Removed old manual button-based source filter UI and logic for displaying individual source buttons
- Updated @dust-tt/sparkle package to version 0.4.24 to include DropdownMenuFilters component updates

tdraier commented on front/components/assistant/conversation/input_bar/InputBarAttachmentsPicker.tsx:
```diff
@@ -446,29 +454,19 @@ export const InputBarAttachmentsPicker = ({
       >
         {searchQuery ? (
           <div ref={itemsContainerRef}>
```
like : <img width="196" height="133" alt="Image" src="https://github.com/user-attachments/assets/e7ad5eba-904e-4a0b-bc68-34883f001aa4" />

JulesBelveze committed:
[front] - refactor: improve attachments picker layout with filters

 - Wrapped loader and filters within a flex container to manage layout more effectively
 - Hardcoded a test filter within the DropdownMenuFilters for debugging purposes
